### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by ixboy**
Hello, this is my To-Do List project, this is my third milestone on this project.

What I have done: hammer

    Created tasks-handler.js file for new methods.
    Added addTask() function to add new tasks.
    Added updateDescription() function to update the content description.
    Added deleteTodo() function to remove tasks.
    Added updateIndex() function to update the index numbers after I delete tasks.
    Added function to clear all completed tasks.
    Added localStorage functions to handle local data.
  Refactored my code and changed some functions into class methods

I have implemented all the requirements, please do review my project and tell me if I can proceed to the next one.
Thank You :)

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,45 +1,79 @@
...
+      if (list.id == id) list.description = newValue;
+    });
+    updateLocalStorage(todos);
+  }
+
+  static deleteTodo(id) {
+    const ind = todos.findIndex((obj) => obj.id == id);
+    todos.splice(ind, 1);
+    updateLocalStorage(todos);
+  }
```
Comment (MahmoudBakr23) [inline src/main.js:66]: - [x] Kindly make sure when an item is deleted, the list of todos is resorted correctly by index.
For example, we have one, two, and three todo items in the list, one has an index = 0, two = 1, and three = 2, and we deleted number two on the list (index = 1), the 3rd item should have an index = 1.
>Please go ahead and check out the project requirements for this point right [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should,saved%20in%20local%20storage.).

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @ixboy 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct
- Descriptive PR summary and commit messages.
- Well done implementing the features of the app
- Nice design

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [x] Please write a more descriptive PR headline to make it easier for viewers and reviewers to quickly catch up on what this PR is about! 😄 🙏 _Kindly keep in mind that this issue would normally get your PR invalidated._

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [x] It'd be super nice if you attach an image of your app to the README file. This makes your repo more professional. 😄 🙏
_Please note that the README file isn't loaded for inline comments._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/main.js (hunk 1 above)
Comment (ixboy) [reply src/main.js:66]: Hello @MahmoudBakr23  Thank You for the feedback!, I really appreciated, however I do have a `function updateIndex() ` in my **index.js** file and I do call it inside the `function updateLocalStorage` it's job is to change the index value inside the to-do list or the item object. And I have tested it, it's working on my side.

Thank You Again for the review.

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @ixboy 😃,

Thank you for your great communication and for making the changes. However, kindly update the PR headline as I mentioned earlier 😄 🙏

Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/main.js (hunk 1 above)
Comment (MahmoudBakr23) [reply src/main.js:66]: Thank you for communicating with me ❤️

**Comment by ixboy** [comment]
Thank You, @MahmoudBakr23   I just updated the PR headline.

Best regards :)
Ismael

**Pull request merged**